ecs-agent/api/container/restart/restart_tracker.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package restart
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	api "github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+type RestartTracker struct {
+	RestartCount  int       `json:"restartCount,omitempty"`
+	LastStartedAt time.Time `json:"lastStartedAt,omitempty"`
+	restartPolicy *api.ContainerRestartPolicy
```

Comment:
Let's create a non-ACS model struct type for restart policy which doesn't use pointer types for all of its fields as this is what was decided during internal design. That way we also avoid messy lines such as L87. 

For example,
```
if time.Since(startedAt) < rt.restartPolicy.AttemptResetPeriod
```
looks a lot cleaner than

```
if time.Since(startedAt) < time.Duration(*rt.restartPolicy.RestartAttemptPeriod*int64(time.Second))
```

This is common practice in Agent code, see environment file struct type for example:
https://github.com/aws/amazon-ecs-agent/blob/dev/ecs-agent/acs/model/ecsacs/api.go#L1074-L1080
https://github.com/aws/amazon-ecs-agent/blob/dev/agent/api/container/container.go#L355-L358